### PR TITLE
added addReservation, updateReservation and updated dele and check en…

### PR DESCRIPTION
**Pull request by AnitaTudor**
## Description
This PR adds the addReservation and updateReservation endpoints, as well as updating the checkReservation to now check the reservation by the name it was made, rather than the unique generated string.Same scenario with deleteReservation endpoint.

### CODE REVIEW

**Review by OlteanuCosmin — APPROVED** [review]

File: src/main/java/com/javafrenzy/restaurant/controller/ReservationController.java [hunk 1]
```diff
@@ -25,16 +22,27 @@
...
+    @ResponseBody
+    public Reservation addReservation(@RequestBody Reservation reservation) {
+        return reservationService.addReservation(reservation);
+    }
+
+    @PutMapping("/updateReservation")
+    @ResponseBody
+    public Reservation updateReservation(@RequestBody Reservation reservation){
+        return reservationService.updateReservation((reservation));
+    }
```
Comment (vladpirlog) [inline src/main/java/com/javafrenzy/restaurant/controller/ReservationController.java:47]: nit: double round brackets `((reservation))`

File: src/main/java/com/javafrenzy/restaurant/exception/ReservationAlreadyAddedException.java [hunk 2]
```diff
@@ -0,0 +1,7 @@
+package com.javafrenzy.restaurant.exception;
+
+public class ReservationAlreadyAddedException extends RuntimeException{
+    public ReservationAlreadyAddedException(String name){
+        super(name+ " already has a reservation!. Try updating it!");
+    }
```
Comment (vladpirlog) [inline src/main/java/com/javafrenzy/restaurant/exception/ReservationAlreadyAddedException.java:6]: nit: blank after `name`

File: src/main/java/com/javafrenzy/restaurant/controller/ReservationController.java (hunk 1 above)
Comment (AnitaTudor) [reply src/main/java/com/javafrenzy/restaurant/controller/ReservationController.java:47]: resolved

File: src/main/java/com/javafrenzy/restaurant/exception/ReservationAlreadyAddedException.java (hunk 2 above)
Comment (AnitaTudor) [reply src/main/java/com/javafrenzy/restaurant/exception/ReservationAlreadyAddedException.java:6]: resolved

**Pull request merged**